### PR TITLE
Stats geotiff

### PR DESCRIPTION
**Pull request by weiqin61**
add the stats function and test in laser3d_stats.

### CODE REVIEW

File: terraref/laser3d/laser3d_stats.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+from rpy2.robjects import pandas2ri
+
+
+class GeotiffStats(object):
+    def __init__(self, file_path):
+        self.file_path = file_path
+        self.file = gdal.Open(file_path)
+        self.vector = np.array(self.file.GetRasterBand(1).ReadAsArray())
+
+    def sample_geotiff(self, sample_num = 1000):
```
Comment (dlebauer) [inline terraref/laser3d/laser3d_stats.py]: We need to make sure that the statistics are independent of spatial autocorrelation. Perhaps we should have a quasi-random sample? I expect that the spatial autocorrelation will have a lag at the scale of leaf structure (<10cm?).

Some of the statistics can be more reliably estimated from a sample of a given size (e.g. mean can be more accurately estimated than variance or quantiles from a sample of 1000).

**Comment by weiqin61** [comment]
sample histogram of temperature geotiff
samplehist

**Comment by weiqin61** [comment]
Currently we face the issue of distinguishing of leaf angle and ground, the leaf angle would only require leaf angle information while ground would need height value (z), which is now in the current leaf angle geotiff. We might think of using las point cloud to do the leaf angle fit in the future, and we would come back to this issue after laser3d package is done.

**Comment by weiqin61** [comment]
I have merged the changes in master to my branch, and now the PR is good to merge.

**Pull request merged**